tests/jax/integration/executor_test.py

```diff
@@ -13,265 +13,130 @@
     (
         data_server,
         parameter_server,
         executor,
         evaluator,
         trainer,
-    ) = test_executor_parameter_server_system._builder.store.system_build
-
-    assert isinstance(executor, acme.core.Worker)
-
-    # Save the executor policy
+    ) = test_system_sp._builder.store.system_build
 
-    parameters = executor._executor.store.executor_parameter_client._parameters
-
-    # Change a variable in the policy network
-    parameter_server.set_parameters(
-        {"evaluator_steps": np.full(1, 1234, dtype=np.int32)}
-    )
-
-    # Step the executor
+    # Remove adder
+    executor._executor.store.adder = None
+    # Run an episode
     executor.run_episode()
 
-    # Check if the executor variable has changed.
-    parameters = executor._executor.store.executor_parameter_client._parameters
-    assert parameters["evaluator_steps"] == 1234
-
-
-#########################################################################
-# Integration test for the executor, adder, data_server, variable_client
-# and variable_server.
-class TestSystemExceptTrainer(System):
-    def design(self) -> Tuple[DesignSpec, Dict]:
-        """Mock system design with zero components.
-
-        Returns:
-            system callback components
-        """
-        components = DesignSpec(
-            **system_init,
-            data_server=OnPolicyDataServer,
-            data_server_adder_signature=ParallelSequenceAdderSignature,
-            extras_spec=ExtrasLogProbSpec,
-            parameter_server=DefaultParameterServer,
-            executor_parameter_client=ExecutorParameterClient,
-            **executor,
-            distributor=Distributor,
-            trainer_parameter_client=mocks.MockTrainerParameterClient,
-            logger=mocks.MockLogger,
-            trainer=mocks.MockTrainer,
-            trainer_dataset=mocks.MockTrainerDataset,
-        )
-        return components, {}
-
-
-@pytest.fixture
-def test_system_except_trainer() -> System:
-    """Add description here."""
-    return TestSystemExceptTrainer()
-
-
-# Skip failing test for now
-@pytest.mark.skip
-def test_except_trainer(
-    test_system_except_trainer: System,
-) -> None:
-    """Test if the parameter server instantiates processes as expected."""
-
-    # Environment.
-    environment_factory = functools.partial(
-        debugging_utils.make_environment,
-        env_name="simple_spread",
-        action_space="discrete",
+    # Observe first (without adder)
+    assert executor._executor.store.adder is None
+
+    # Select actions and select action
+    assert list(executor._executor.store.actions_info.keys()) == [
+        "agent_0",
+        "agent_1",
+        "agent_2",
+    ]
+    assert list(executor._executor.store.policies_info.keys()) == [
+        "agent_0",
+        "agent_1",
+        "agent_2",
+    ]
+    num_possible_actions = environment_factory().action_spec()["agent_0"].num_values
+    assert (
```

Comment:
Imo lambda is not ideal for readibilty but more imp, isn't this assertion going to check for every agent, its action matches the possible action so agent_0. It would work in simple spread as all agents have same actions but we should change this to check num_possible for each agent

Reply:
Do you have an ideal function to check if the action choosen is within the possible ones another option was thinking to check if the action is within the shape of legal_action since the shape gives an idea about how many action we have

Reply:
```
    num_possible_actions = [
        environment_factory().action_spec()[agent].num_values
        for agent in environment_factory().possible_agents
    ]
    for i in range(len(num_possible_actions)):
        assert list(executor._executor.store.actions_info.values())[i] in range(
            0, num_possible_actions[i]
        )
```

Reply:
Thank you!

Reply:
Updated! Thanks @AsadJeewa